plan/optimizer.go

```diff
@@ -31,7 +31,9 @@ const (
 	flagPrunColumns uint64 = 1 << iota
 	flagEliminateProjection
 	flagBuildKeyInfo
+	flagEliminateAgg
 	flagDecorrelate
+	flagEliminateProjection2
 	flagMaxMinEliminate
```

Comment:
Any reason why flagEliminateProjection2 is applied after flagDecorrelate?

Reply:
Seems i can just place it after the `ElimiateAgg` before the `Decorrelate`. This will be done in the newly coming commit.